### DISCUSSION TITLE
Long term archive of a specific vcpkg build environment?

### DISCUSSION THREAD

**Discussion by hajokirchhoff** [microsoft/vcpkg#22427] [Q&A]
Hi there,
I really like the way vcpkg has developed in the last couple of years. One thing bothers me, though: vcpkg install <somelib> will always download sources from the internet when it needs to build a lib not yet installed or cached. This is no problem now, but what about in 10 or 15 years from now?

This is not a theoretical question, quite the contrary. Last November I had to fix a bug in a software last released in 2009 - 12 years old. This was no problem, because the entire (!) build environment was checked into an svn repositry. All I needed was to install VS2008, check out the repository and find and fix the bug. I always make sure that _all_ 3rd party dependencies are checked into our repositories.

What is the suggested way to do this with vcpkg? Pushing the specific vcpkg version used into our git repository is obviously not enough. The servers where the sourcecode gets downloaded from might no longer exist 12 years from now. OTOH I don't want to check in the entire vcpkg binary buildtrees.

And just adding the directories produced by vcpkg (consumed by the build) is also not enough. I want to be able to debug and patch the original libraries, so the exact and complete library sources and build environment needs to be archived.

Is there a way to export all "downloaded" packages and import them in the future so that vcpkg can rebuild all libraries in the exact same version - but without accessing any external servers?

I could of course simply make a backup of the entire vcpkg tree, but that would include GBs of useless binaries and I would not be able to add this to a git server. Not want I want.

Thanks
Hajo

**Comment by autoantwort** [comment]
To solve this issue we have a local copy of the vcpkg repo and use [asset caching](https://github.com/microsoft/vcpkg/blob/master/docs/users/assetcaching.md) to cache everything else.

**Reply by hajokirchhoff** [reply]
Great pointer, thanks.
